tensorizer/serialization.py

```diff
@@ -2333,16 +2334,20 @@ def _to_torch_parameter(
 
         gradient = tensor.dtype.is_complex or tensor.dtype.is_floating_point
 
-        start = time.perf_counter()
+        start = time.perf_counter_ns() if _perf_stats else 0
         tensor_on_device = tensor.to(device=self._device, dtype=target_dtype)
-        duration = time.perf_counter() - start
-        _perf_stats.cuda_to_device_millisecs.add(int(1000.0 * duration))
+        end = time.perf_counter_ns() if _perf_stats else 0
 
         result = torch.nn.Parameter(
             tensor_on_device,
             requires_grad=gradient,
         )
-        _perf_stats.cuda_bytes.add(result.element_size() * result.nelement())
+        if _perf_stats and original_device != result.device:
+            duration = end - start
+            bytes_transferred = result.element_size() * result.nelement()
+            with _perf_stats.lock:
+                _perf_stats.tensor_to_device_ns += duration
+                _perf_stats.tensor_to_device_bytes += bytes_transferred
```

Comment:
At this point I'm pretty tempted to code the timer itself as a context within _perf_stats to keep the main code cleaner. Can be left as future work tho.